scripts/cmakegen/cmakegen.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+
+#
+#  Copyright 2014-2016 CyberVision, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import sys
+import os
+import yaml
+from string import Template
+
+CMAKE_LISTS_TEMPLATE = open(os.path.join(os.path.dirname(__file__),'CMakeLists.txt')).read()
+EMBEDDED_TEMPLATE = open(os.path.join(os.path.dirname(__file__),'Embedded.cmake')).read()
+POSIX_TEMPLATE = open(os.path.join(os.path.dirname(__file__),'Posix.cmake')).read()
+DEFINITION_TEMPLATE = open(os.path.join(os.path.dirname(__file__),'Definition.cmake')).read()
+
+class CMakeGenTemplate(Template):
+    delimiter = '%'
+
+class GeneratorException(Exception):
+    pass
+
+class FeaturesGenerator:
+    def __init__(self, features):
+        self._features = list(features) if features is not None else []
+
+    def generate(self, language):
+        if self._features is None:
+            return str()
+        template_c = 'set(WITH_EXTENSION_%extension OFF)\n'
+        template_cpp = 'set(KAA_WITHOUT_%extension 1)\n'
+        if language == 'c':
+            return self._generate_from_template(template_c)
+        elif language == 'cpp':
+            return self._generate_from_template(template_cpp)
+        else:
+            raise GeneratorException('Unknown language: '+language)
+
+    def _generate_from_template(self, template):
+        features = ['configuration', 'notifications', 'logging', 'events']
+        output = ''
+        for f in features:
+            if not f in self._features:
+                output = output + CMakeGenTemplate(template).substitute(extension = f.upper())
+        return output
+
+
+class DefinitionsGenerator:
+    def __init__(self, definitions):
+        self._definitions = dict(definitions) if definitions is not None else dict()
+
+    def add_definitions(self, **kwargs):
+        self._definitions.update(kwargs)
+
+    def generate(self):
+        out = str()
+        for k, v in self._definitions.iteritems():
+            out = out + CMakeGenTemplate(DEFINITION_TEMPLATE).safe_substitute(variable=k, value=v)
+        return out
+
+class Generator:
+    def __init__(self, basedir, **kwargs):
+        self._appname = str(kwargs.get('appname'))
+        self._path = os.path.abspath(os.path.join(basedir,kwargs.get('path')))
+        self._languages = list(kwargs.get('languages'))
+        self._build_embedded = kwargs.get('build_embedded')
+        self._features = list(kwargs.get('features') or [])
```

Comment:
BTW, you can use the default argument to the get() method:

_self._features = list(kwargs.get('features', []))_